src/poetry/vcs/git/backend.py

```diff
@@ -0,0 +1,396 @@
+        }:
+            local.refs.import_refs(
+                base=base,
+                other={
+                    n[len(prefix) :]: v
+                    for (n, v) in remote_refs.refs.items()
+                    if n.startswith(prefix) and not n.endswith(ANNOTATED_TAG_SUFFIX)
+                },
+            )
+
+        try:
+            with local:
+                local.reset_index()
+        except (AssertionError, KeyError) as e:
+            # this implies the ref we need does not exist or is invalid
+            if isinstance(e, KeyError):
+                # the local copy is at a bad state, lets remove it
+                safe_rmtree(local.path, ignore_errors=True)
+
+            if isinstance(e, AssertionError) and "Invalid object name" not in str(e):
+                raise
+
+            raise PoetrySimpleConsoleException(
+                f"Failed to clone {url} at '{refspec.key}'"
+            )
+
+        return local
+
+    @classmethod
+    def _clone_submodules(cls, repo: Repo) -> None:
+        """
+        Helper method to identify configured submodules and clone them recursively.
+        """
+        repo_root = Path(repo.path)
+        modules_config = repo_root.joinpath(".gitmodules")
+
+        if modules_config.exists():
+            config = ConfigFile.from_path(modules_config)
+
+            url: bytes
+            path: bytes
+            for path, url, _ in parse_submodules(config):
+                path_relative = Path(path.decode("utf-8"))
+                path_absolute = repo_root.joinpath(path_relative)
+
+                source_root = path_absolute.parent
+                source_root.mkdir(parents=True, exist_ok=True)
+
+                with repo:
+                    revision = repo.open_index()[path].sha.decode("utf-8")
+
+                cls.clone(
+                    url=url.decode("utf-8"),
+                    source_root=source_root,
+                    name=path_relative.name,
+                    revision=revision,
+                    clean=path_absolute.exists()
+                    and not path_absolute.joinpath(".git").is_dir(),
+                )
+
+    @staticmethod
+    def is_using_legacy_client() -> bool:
+        from poetry.factory import Factory
+
+        return (
+            Factory.create_config()
+            .get("experimental", {})
+            .get("system-git-client", False)
+        )
+
+    @staticmethod
+    def get_default_source_root() -> Path:
+        from poetry.factory import Factory
+
+        return Path(Factory.create_config().get("cache-dir")) / "src"
+
+    @classmethod
+    def clone(
+        cls,
+        url: str,
+        name: str | None = None,
+        branch: str | None = None,
+        tag: str | None = None,
+        revision: str | None = None,
+        source_root: Path | None = None,
+        clean: bool = False,
+    ) -> Repo:
+        if not source_root:
+            source_root = cls.get_default_source_root()
```

Comment:
Just a personal taste:

```suggestion
        source_root = source_root or cls.get_default_source_root()
```

Some line below you are doing this as well with `name`.

Reply:
Sure, I suspect additional logic got stripped away.